aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -196,10 +264,11 @@ public TransferObserver upload(String bucket, String key, File file, ObjectMetad
      * @param file The file to upload.
      * @param metadata The S3 metadata to associate with this object
      * @param cannedAcl The canned ACL to associate with this object
+     * @param connectionCheckType Type of connection check. Default is {@link NetworkInfoReceiver.Type#WIFI_ONLY}
      * @return A TransferObserver used to track upload progress and state
      */
     public TransferObserver upload(String bucket, String key, File file, ObjectMetadata metadata,
-            CannedAccessControlList cannedAcl) {
+                                   CannedAccessControlList cannedAcl, NetworkInfoReceiver.Type connectionCheckType) {
```

Comment:
can we instead move the WAN configuration into some sort of a TransferConfiguration class? that way API doesnt have to change much between feature changes.